### ISSUE TITLE
PhantomJS: "undefined is not a constructor"

### ISSUE THREAD

**Issue by dustinfarris** [mattkrick/cashay#136]
I'm running acceptance tests in Ember.  The tests pass in Chrome, however when I run the tests in Phantom JS, I am getting this error:

```
TypeError: undefined is not a constructor (evaluating 'type.startsWith('@@cashay')')
```

which seems to stem from this line:

https://github.com/mattkrick/cashay/blob/27320c4ebd5c5d2ce697479d21a72de7f3d7c73c/src/normalize/duck.js#L38

Any idea what might cause this?

**Comment by mattkrick** [comment]
Did you set up the redux store for your tests?

**Comment by dustinfarris** [comment]
Think I figured it out.  It is because PhantomJS does not support `String.prototype.startsWith`.  I am trying to find a polyfill I can inject.

**Comment by mattkrick** [comment]
Whoa! Thats interesting. I could use indexOf, startsWith is just a little
more expressive

**Comment by dustinfarris** [comment]
Here's a polyfill you could add to cashay's dependencies: https://github.com/mathiasbynens/String.prototype.startsWith

**Comment by dustinfarris** [comment]
But now getting the same thing for Object.prototype.find

e.g.: https://github.com/mattkrick/cashay/blob/8d9174d478b63883f68b64f82c183b60a3aad0c4/src/normalize/denormalizeStore.js#L56

This may be something I as a user should be responsible for polyfilling (I wonder if babel polyfill would make all of this go away).

**Comment by dustinfarris** [comment]
Yeah, running everything through babel polyfill makes this all go away.

This can be closed as far as I'm concerned, unless there is any docs action you think would be appropriate.

**Comment by dustinfarris** [comment]
I am hitting this again, this time in plain-ol node

```js
var transformSchema = require('cashay').transformSchema;

transformSchema(myRootSchema...
```

results in 

```
TypeError: undefined is not a function
    at /Users/dustin/Archive/broccoli-cashay-schema/node_modules/cashay/lib/schema/transformSchema.js:127:23
```

stemming from: https://github.com/mattkrick/cashay/blob/master/src/schema/transformSchema.js#L39

**Comment by mattkrick** [comment]
ah, now that we don't have the polyfill, it doesn't have `startsWith`?

**Comment by dustinfarris** [comment]
Correct—if I install/require the polyfill mentioned above everything works.  What do you think about doing this in cashay instead?

Add [String.prototype.startsWith](https://github.com/mathiasbynens/String.prototype.startsWith) to `dependencies` and in index.js:

```js
import 'string.prototype.startsWith';
```

(not sure how the dots in the name will affect `import`)

**Comment by dustinfarris** [comment]
and/or do it with babel-polyfill might be better

**Comment by dustinfarris** [comment]
Well the fact that I was using node v0.12.6 probably wasn't helping 👎 

What minimum version of node do we want to support?

**Comment by mattkrick** [comment]
I'd say 6 LTS, although I think 4 would work
